### DISCUSSION TITLE
about the gradle script of root project

### DISCUSSION THREAD

**Discussion by buffcow** [LSPosed/LSPosed#2770] [Q&A] [answered]
It is so great to see [this global configuration script](https://github.com/LSPosed/LSPosed/blob/master/build.gradle.kts) you wrote @yujincheng08 . 
But I encountered a problem about configuring kotlinOptions, how do I configure kotlinOptions in the global script?
in sub project, i can configure it like this:
```kotlin
android{
    ...
    kotlinOptions {
        jvmTarget = "1.8"
        javaParameters = true
    }
    ...
}
```

kotlinOptions seems to be an extension function of BaseAppModuleExtension, but I cannot reference it in the global configuration. What should I do😢?

**Comment by tianjyan** [comment, marked as answer]
try this in your root build.gradle.kts
```kts
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile

...

tasks.withType(KotlinCompile::class.java).all {
    kotlinOptions.jvmTarget = "1.8"
    kotlinOptions.javaParameters = true
}
```

**Reply by buffcow** [reply]
so cool, it totally works now😁,thank you very much~
